app/config/cache.js

```diff
@@ -0,0 +1,79 @@
+const Joi = require('joi')
+const { serviceName } = require('../constants/service-name')
+const { DEVELOPMENT, TEST, PRODUCTION } = require('../constants/environments')
+const { DAY } = require('../constants/time')
+
+const REDIS_PORT = 6379
+
+const schema = Joi.object({
+  host: Joi.string(),
+  port: Joi.number().integer().default(REDIS_PORT),
+  password: Joi.string().allow(''),
+  partition: Joi.string().default(serviceName),
+  cacheName: Joi.string().default(serviceName),
+  ttl: Joi.number().integer().default(DAY), // 24 hours
+  tls: Joi.object(),
+  segment: Joi.string().default(serviceName),
+  expiresIn: Joi.number().default(DAY) // 24 hours
+})
+
+const config = {
+  host: process.env.REDIS_HOST,
+  port: process.env.REDIS_PORT,
+  password: process.env.REDIS_PASSWORD,
+  partition: process.env.REDIS_PARTITION ?? process.env.SERVICE_NAME,
+  cacheName: process.env.REDIS_CACHE_NAME ?? process.env.SERVICE_NAME,
+  ttl: process.env.REDIS_TTL,
+  segment: process.env.REDIS_API_SEGMENT,
+  tls: process.env.NODE_ENV === PRODUCTION ? {} : undefined
+}
+
+const result = schema.validate(config, {
+  abortEarly: false
+})
+
+if (result.error) {
+  throw new Error(`The cache config is invalid. ${result.error.message}`)
+}
+
+const value = result.value
+
+value.isDev = process.env.NODE_ENV === DEVELOPMENT
+value.isTest = process.env.NODE_ENV === TEST
+value.isProd = process.env.NODE_ENV === PRODUCTION
+
+value.useRedis = value.host !== undefined
+
+if (!value.useRedis) {
+  console.info('Redis disabled, using in memory cache')
+}
```

Comment:
Very minor but would be nice to have a log entry if redis is enabled so we can easily see when deployed to say DEV or TEST that the config thinks it's running